pages/feed/index.tsx

```diff
@@ -179,7 +186,18 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
                         addSuffix: true,
                       })}
                     </span>
-                    <DialogCloseButton onClick={() => router.push("/feed")} />
+                    {props.referer !== null && !props.referer.includes("/feed") ? (
+                      <Button
+                        variant="text"
+                        onClick={() => router.back()}
+                        className="!p-0 !border-0 absolute top-4 right-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-slate-400 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-slate-100"
+                      >
+                        <AiOutlineClose size={20} />
+                        <span className="sr-only">Close</span>
+                      </Button>
+                    ) : (
+                      <DialogCloseButton onClick={() => router.replace("/feed")} />
+                    )}
```

Comment:
Why did not we have the condition inside this code
```js
 <DialogCloseButton onClick={() => {
 let isOnFeedPage = props.referer !== null && !props.referer.includes("/feed") 
 isOnFeedPage ? router.replace("/feed") :  router.back();  
 }} />
```

Reply:
That's because of the `DialogCloseButton` default behaviour, It closes the `Dialog`.

It does that before anything else i.e. it will do that before running (or even in the absence of) the `onClick` function, hence creating a scenario where the `/feed` page in the background displays first before the routing takes place at all.

So in the case where `referer` is not the '/feed' page, it still displays the '/feed' page first before routing back to the `referer` and the main reason for the PR is to return to `referer` without closing dialog to the '/feed' page.

Reply:
It is important to state that it is the Primitive function of the `DialogCloseButton` to close the `Dialog`, this means that it is the button that closes the `Dialog` by itself and not the `onClick` function.

The function of the `onClick={() => router.replace("/feed")}` is to just update the current page/route url when the `Dialog` closes without adding another entry to the browser `history`, its more like a callback to the close action.